pandas/core/apply.py

```diff
@@ -410,17 +410,32 @@ def agg_dict_like(self) -> DataFrame | Series:
             context_manager = com.temp_setattr(obj, "as_index", True)
         else:
             context_manager = nullcontext()
+
+        if isinstance(selected_obj, ABCDataFrame):
```

Comment:
Below can you just check `if selected_obj.ndim == 2 and selected_obj.columns.nunique() < len(selected_obj.columns)`?